### PR TITLE
ACA 137: Ability to identify the Jurisdiction of a Solicitor

### PR DESCRIPTION
**Pull request by tomxelliott**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ACA-137

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java [hunk 1]
```diff
@@ -65,8 +67,9 @@ private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
 
-    public boolean hasSolicitorRole(List<String> roles) {
-        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches());
+    public boolean hasSolicitorRoleForJurisdiction(List<String> roles, String jurisdiction) {
+        String jurisdictionRole = String.format(CASEWORKER_JURISDICTION, jurisdiction).toLowerCase(Locale.getDefault());
+        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches()
+            && role.toLowerCase(Locale.getDefault()).startsWith(jurisdictionRole));
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java:73]: The logic is slightly different to data-store-api but works fine as long as below multi jurisdiction scenario never exist. Could you check with Richard once please. 
role : `caseworker-divorce-financialremedy-solicitor` but jurisdiction : financialremedy ??

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java:73]: Will check with Richard. I had thought that the jurisdiction in the above example was divorce?

File: src/test/java/uk/gov/hmcts/reform/managecase/security/SecurityUtilsTest.java [hunk 2]
```diff
@@ -38,30 +39,30 @@ public void setup() {
...
         roles.add("caseworker-befta_jurisdiction-barrister");
-        assertFalse(securityUtils.hasSolicitorRole(roles));
+        assertFalse(securityUtils.hasSolicitorRoleForJurisdiction(roles, JURISDICTION));
     }
 
     @Test
     void hasSolicitorRoleReturnsFalseWithAdditionalSuffixAppended() {
         roles.add("caseworker-befta_jurisdiction-solicitorsurname-solicitor-role");
-        assertFalse(securityUtils.hasSolicitorRole(roles));
+        assertFalse(securityUtils.hasSolicitorRoleForJurisdiction(roles, JURISDICTION));
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/managecase/security/SecurityUtilsTest.java:66]: **[As per #223]** Should there at least be a jurisdiction negative test now that jurisdictions have been re-added to the function? Also may be worth adding a positive test using the failing format `caseworker-ia-legalrep-solicitor` that created ACA-119.

**Review by mattnayler — COMMENTED** [review]
Two minor review comments added that correspond to similar in #223.

**Review by smathangi — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/managecase/api/controller/CaseAssignmentControllerIT.java [hunk 3]
```diff
@@ -67,8 +67,8 @@
     private static final String ANOTHER_USER = "vcd345cvs-816a-4eea-b714-6654d022fcef";
     private static final String CASE_ID = "1588234985453946";
     private static final String CASE_ID2 = "1598630369818638";
-    private static final String ORG_POLICY_ROLE = "caseworker-probate";
-    private static final String ORG_POLICY_ROLE2 = "caseworker-probate2";
+    private static final String IDAM_USER_ROLE = "caseworker-probate-solicitor";
+    private static final String IDAM_USER_ROLE2 = "caseworker-probate2-solicitor";
```
Comment (mattnayler) [inline src/integrationTest/java/uk/gov/hmcts/reform/managecase/api/controller/CaseAssignmentControllerIT.java:71]: see https://github.com/hmcts/aac-manage-case-assignment/pull/223#discussion_r567789833

File: src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java [hunk 4]
```diff
@@ -65,8 +66,8 @@ private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
 
-    public boolean hasSolicitorRole(List<String> roles) {
-        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches());
+    public boolean hasSolicitorRoleForJurisdiction(List<String> roles, String jurisdiction) {
+        return roles.stream().anyMatch(role -> SOLICITOR_ROLE.matcher(role).matches()
+            && role.toLowerCase(Locale.getDefault()).contains(jurisdiction.toLowerCase(Locale.getDefault())));
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/managecase/security/SecurityUtils.java:71]: see https://github.com/hmcts/aac-manage-case-assignment/pull/223#discussion_r567784045

**Review by mattnayler — CHANGES_REQUESTED** [review]
See matching comments on #223.  Other changes appreciated, thanks.

**Review by mattnayler — APPROVED** [review]
LGTM 👍

**Review by smathangi — APPROVED** [review]

**Pull request merged**